lib/site_setting_extension.rb

```diff
@@ -200,41 +200,50 @@ def client_settings_json
   end
 
   def client_settings_json_uncached
-    MultiJson.dump(
-      Hash[
-        *@client_settings
-          .flat_map do |name|
-            # Themeable site settings require a theme ID, which we do not always
-            # have when loading client site settings. They are excluded here,
-            # to get them use theme_site_settings_json(:theme_id)
-            next if themeable[name]
-
-            value =
-              if deprecated_settings.include?(name.to_s)
-                public_send(name, warn: false)
-              else
-                public_send(name)
-              end
-
-            type = type_supervisor.get_type(name)
-            if type == :upload
-              value = value.to_s
-            elsif type == :uploaded_image_list
-              value = value.map(&:to_s).join("|")
-            end
-
-            [name, value]
+    uncached_json =
+      @client_settings.filter_map do |name|
+        # Themeable site settings require a theme ID, which we do not always
+        # have when loading client site settings. They are excluded here,
+        # to get them use theme_site_settings_json(:theme_id)
+        next if themeable[name]
```

Comment:
Seems like this line is causing the issue in the old `flat_map` + `compact` way of doing things, but not sure _why_ exactly -- some interaction with our internal plugin. It works now after trying the internally failing specs with this fix